cmd/prometheus/main.go

```diff
@@ -1180,6 +1180,7 @@ func main() {
 	if agentMode {
 		// WAL storage.
 		opts := cfg.agent.ToAgentOptions()
+		opts.SetOutOfOrderTimeWindow(cfg.tsdb.OutOfOrderTimeWindow)
```

Comment:
Hmmm, maybe we could remove this new line and set the out-of-order window inside `ToAgentOptions`?

Not sure if that's nitpicking or not to be honest, what do you think?

Reply:
Done. Is that what you had in mind?